File: backend/src/main/java/com/a603/ofcourse/domain/place/dto/PlaceDto.java
package com.a603.ofcourse.domain.place.dto;

import com.a603.ofcourse.domain.place.domain.Place;
import com.a603.ofcourse.global.common.Points;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Getter
public class PlaceDto {
    private Integer id;
    private String name;
    private String address;
    private String url;
    private String imageUrl;
    private String placeType;
    private String placeCategory;
    private Points points;

    public static PlaceDto of(Place place) {
        return new PlaceDto(
                place.getId(),
                place.getName(),
                place.getAddress(),
                place.getUrl(),
                place.getImageUrl(),
                place.getPlaceType(),
                place.getPlaceCategory().getValue(),
                place.getPoints()
        );
    }
}

File: dto/place_dto.py
from pydantic import BaseModel

from domain.place import Place


class PlaceDto(BaseModel):
    place_id: int
    name: str
    address: str
    url: str
    restaurant_type: str
    place_category: str
    latitude: float
    longitude: float

    @classmethod
    def from_place(cls, place: Place) -> 'PlaceDto':
        return cls(
            place_id=place.place_id,
            name=place.name,
            address=place.address,
            url=place.url,
            restaurant_type=place.restaurant_type,
            place_category=place.place_category,
            latitude=place.latitude,
            longitude=place.longitude,
        )
